### PR TITLE
Mirage xen/solo5 api change

### PR DESCRIPTION
**Pull request by TheLortex**
Test the first step of API changes in mirage-solo5, mirage-xen. 
This makes duniverse fully build with opam (according to CI)

### CODE REVIEW

File: packages/mirage-bootvar-xen/mirage-bootvar-xen.0.5.0/opam [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+name: "mirage-bootvar-xen"
+maintainer: "Magnus Skjegstad <magnus@skjegstad.com>"
+authors: "Magnus Skjegstad <magnus@skjegstad.com>"
+homepage: "https://github.com/mirage/mirage-bootvar-xen"
+bug-reports: "https://github.com/mirage/mirage-bootvar-xen/issues/"
+dev-repo: "git+https://github.com/TheLortex/mirage-bootvar-xen.git"
+license: "ISC"
+
+build: [
+  [ "jbuilder" "subst" "-p" name ] {pinned}
```
Comment (avsm) [inline packages/mirage-bootvar-xen/mirage-bootvar-xen.0.5.0/opam]: jbuilder?? :-)

Comment (TheLortex) [reply packages/mirage-bootvar-xen/mirage-bootvar-xen.0.5.0/opam]: Looks like mirage-bootvar-xen is still using jbuild files !

**Comment by Julow** [comment]
The CI fail maybe because it tries to build examples folders. Also there is a lot of:
```
 ppxfind duniverse/ppx_deriving/src_plugins/create/ppx_deriving_create.pp.ml (exit 2) 
(cd _build/default && ../install/default/bin/ppxfind -legacy ppx_tools.metaquot --as-pp duniverse/ppx_deriving/src_plugins/create/ppx_deriving_create.ml) > _build/default/duniverse/ppx_deriving/src_plugins/create/ppx_deriving_create.pp.ml 
Fatal error: exception Failure("Config file not found - neither /home/opam/.opam/4.07/lib/ocaml/findlib.conf nor the directory /home/opam/.opam/4.07/lib/ocaml/findlib.conf.d") 
```

**Comment by avsm** [comment]
The duniverse build will definitely fail -- it's just worth trying to figure out why some of the errors happen. (for example, the ppxfind invocation is because its calling ocamlfind directly, which we would rather avoid in the duniverse)

**Pull request merged**